lib/tooling/base-tool.js

```diff
@@ -45,6 +45,12 @@ export class BaseTool {
         return this.tool.type || 'independent';
     }
 
+    getUniqueFilePrefix() {
+        const timestamp = process.hrtime();
```

Comment:
Elsewhere we use the `temp` [library](https://www.npmjs.com/package/temp) to do this kind of stuff; but that doesn't seem to want to work with arbitrary directories... and I looked at its implementation and you're basically doing the same thing! So, this is cool :)